File: services/delivery-service/src/main/kotlin/com/pl/platform/svc/messaging/adapter/publisher/EventToTopic.kt
package com.pl.platform.svc.messaging.adapter.publisher
import com.pl.platform.svc.delivery.application.event.DeliveryEventType
import org.springframework.stereotype.Component

@Component
class EventToTopic {
    fun map(eventType: String, module: String): String {
        return when (eventType) {
            DeliveryEventType.CONFIRMED.value -> DeliveryEventType.CONFIRMED.value
            DeliveryEventType.COMPLETED.value -> DeliveryEventType.COMPLETED.value
            else -> "$module.events"
        }
    }
}

File: dotnet/PaymentService/Delivery.Platform.PaymentService/Publisher/EventToTopic.cs
using Delivery.Platform.PaymentService.Events.Contracts;

namespace Delivery.Platform.PaymentService.Publisher;

public static class EventToTopic
{
    public static string Map(string eventType, string module)
    {
        List<string> ownTopics =
        [
            PaymentEventNames.Paid,
            PaymentEventNames.Declined,
            PaymentEventNames.Completed
        ];

        return ownTopics.Contains(eventType) ? eventType : $"{module}.events";
    }
}